.github/workflows/ci.yml

```diff
@@ -41,24 +27,27 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-
-      - name: Setup Bun
-        uses: oven-sh/setup-bun@v1
         with:
-          bun-version: latest
+          ref: ${{ github.event.pull_request.head.ref }}
 
-      - name: Cache Bun dependencies
-        uses: actions/cache@v4
+      - uses: ./.github/actions/setup
         with:
-          path: ~/.bun/install/cache
-          key: ${{ runner.os }}-bun-${{ hashFiles('**/bun.lockb') }}
-          restore-keys: |
-            ${{ runner.os }}-bun-
+          os: ${{ runner.os }}
 
-      - name: Install dependencies
+      - name: Lint fix
```

Comment:
I think fixes don't always solve all the problems, so it may be a good idea to just let the developers run lint fix before they push, and on GitHub Actions we just check if it aligns with the style